sdks/python/apache_beam/runners/direct/test_stream_impl.py

```diff
@@ -226,17 +237,53 @@ def expand(self, pcoll):
   def _infer_output_coder(self, input_type=None, input_coder=None):
     return self.coder
 
-  def _events_from_script(self, index):
-    yield self._events[index]
-
-  def events(self, index):
-    return self._events_from_script(index)
-
-  def begin(self):
-    return 0
-
-  def end(self, index):
-    return index >= len(self._events)
+  @staticmethod
+  def events_from_script(events):
+    """Yields the in-memory events.
+    """
+    return itertools.chain(events)
```

Comment:
Let's add an assert to make sure that this is only called when we have in-memory events vs rpc events (also in the method below).

Reply:
I added an assert in the TestStream constructor so that this shouldn't happen at pipeline run-time but construction time.